python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -98,63 +92,61 @@ def _read_paths(
 
         dataset_kwargs = dataset_kwargs or {}
         dataset_kwargs["partitioning"] = partitioning or "hive"
-        with ExitStack() as stack:
-            # Non-local filesystem handling
-            paths_or_fobs = paths
-            if not _is_local_filesystem(fs):
-                paths_or_fobs = _open_remote_files(
-                    paths_or_fobs,
-                    fs,
-                    context_stack=stack,
-                    **_default_open_file_options(
-                        open_file_options, columns, row_groups
-                    ),
-                )
 
-            # Filter out deprecation warning unless the user
-            # specifies open_file_options and/or use_python_file_object.
-            # Otherwise, the FutureWarning is out of their control.
-            with maybe_filter_deprecation(
-                (
-                    not open_file_options
-                    and "use_python_file_object" not in kwargs
+        # Non-local filesystem handling
+        paths_or_fobs = paths
+        if not _is_local_filesystem(fs):
+            # Use fsspec to transfer byte ranges ahead of time
+            prefetcher = kwargs.pop("prefetcher", "parquet")
+            prefetcher_options = (
+                {
+                    "columns": columns,
+                    # All paths must have the same row-group selection
+                    "row_groups": row_groups[0] if row_groups else None,
+                }
+                if prefetcher == "parquet"
+                else {}
+            )
+            paths_or_fobs = _prefetch_remote_buffers(
+                paths,
+                fs,
+                prefetcher=prefetcher,
+                prefetcher_options=prefetcher_options.update(
+                    kwargs.pop("prefetcher_options", {})
```

Comment:
This can all be replaced with the following code for now (if we don't mind transferring **all** data):

```python
            paths_or_fobs = [
                _fsspec_data_transfer(fpath, fs=fs)
                for fpath in paths
            ]
```

(This wouldn't be as performant, but we also wouldn't need to add any code to `cudf/utils/ioutils.py` for now)